packages/blocks/src/page-block/default/selection-manager.ts

```diff
@@ -420,6 +423,60 @@ export class DefaultSelectionManager {
       hoverEditingState.position.x = hoverEditingState.position.right + 10;
       this._signals.updateCodeBlockOption.emit(hoverEditingState);
     } else {
+      const hoverEditingState = getBlockEditingStateByPosition(
+        this.state.blockCache,
+        e.raw.pageX + 20,
+        e.raw.pageY
+      );
+      if (hoverEditingState?.model) {
+        this._dragHandleAbortController.abort();
+        this._dragHandleAbortController = new AbortController();
+        const currentModel = hoverEditingState.model;
+        const element = getBlockElementByModel(currentModel)!;
+        showDragHandle({
+          anchorEl: element,
+          abortController: this._dragHandleAbortController,
+          onMouseDown: () => {
+            this._setSelectedBlocks([element]);
+          },
+          onDrop: e => {
+            const hoverEditingState = getBlockEditingStateByPosition(
+              this.state.blockCache,
+              e.pageX,
+              e.pageY
+            );
+            if (hoverEditingState) {
+              const nextModel = hoverEditingState.model;
+              const currentParentModel = this.page.getParent(
+                currentModel
+              ) as BaseBlockModel;
+              const nextParentModel = this.page.getParent(
+                nextModel
+              ) as BaseBlockModel;
+              const idx = currentParentModel.children.findIndex(
+                m => m.id === currentModel.id
+              );
+              const nextIdx = nextParentModel.children.findIndex(
+                m => m.id === nextModel.id
+              );
```

Comment:
Looks like all these lines can be moved after the `if (
                recursiveFindParent(this.page, currentModel, nextModel) ||
                recursiveFindParent(this.page, nextModel, currentModel)` check